include/opt-sched/Scheduler/OptSchedTarget.h

```diff
@@ -76,11 +82,12 @@ template <typename FactoryT> class OptSchedRegistry {
 
   FactoryT getFactoryWithName(llvm::StringRef Name) {
     FactoryT Factory = nullptr;
-    for (auto I = List; I; I = I->Next)
-      if (I->Name == Name) {
+    for (auto I = List; I; I = I->Next) {
+      if (strncmp(I->Name.data(), Name.data(), I->Name.size()) == 0) {
```

Comment:
`strncmp`, as a C stdlib function, is easy to use incorrectly and harder to read. I don't see any reason why the `==` doesn't work here. If it's a substring comparison, we should use a string_view library.